### ISSUE TITLE
Wayland support?

### ISSUE THREAD

**Issue by woutervanh** [i-rinat/freshplayerplugin#381]
Would this wrapper also work on wayland?

**Comment by i-rinat** [comment]
Nope. The code is uses Xlib and GTK+ a lot. NPAPI itself is heavily tied to Xlib/X11 and GTK+, with both explicit and implicit dependencies. There is no way to make it work on Wayland.

It's possible to make PPAPI Flash work on Wayland, as PPAPI isolates all interactions with a system, to make sandboxing possible. There was project Mortar[1] at Mozilla. It could allow working PPAPI Flash even on Wayland. I guess it's possible to implement something like that with minimal changes to the Firefox, with all PPAPI implementation living in a browser addon, rather than in Firefox itself, as Mortar had planned.

[1] https://wiki.mozilla.org/Mortar_Project; see page history for the hidden content.

**Comment by woutervanh** [comment]
As tradeof, I try to make it work in xwayland, so wayland which is providing a X11 window to the browser. I have epiphany running that way. However, I still receive an error from flash/fresh:

> NOT SANDBOXED
Vector smash protection is enabled
[fresh] [error] NPP_SetWindow, failed to get NPNVnetscapeWindow
[fresh] [error] NPP_SetWindow, failed to get NPNVnetscapeWindow
[ERROR:flash/platform/pepper/pep_shader_factory.cpp(408)] Compilation failed: 0:1(1): error: syntax error, unexpected Send

[ERROR:flash/platform/pepper/pep_shader_factory.cpp(408)] Compilation failed: 0:1(1): error: syntax error, unexpected Send

sh: gdb: command not found
gdb return status: 32512
